File: src/main/kotlin/com/nuutrai/reactor/listeners/PlayerSwapHand.java
package com.nuutrai.reactor.listeners;

import com.nuutrai.reactor.data.DataManager;
import com.nuutrai.reactor.player.PlayerData;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerSwapHandItemsEvent;

public class PlayerSwapHand implements Listener {

    @EventHandler
    public void onSwapHand(PlayerSwapHandItemsEvent e) {

        e.setCancelled(true);
        PlayerData pd = DataManager.get(e.getPlayer());
        pd.setPaused(!pd.isPaused());

    }

}

File: src/main/kotlin/com/nuutrai/reactor/listeners/PlayerSwapHand.kt
package com.nuutrai.reactor.listeners

import com.nuutrai.reactor.data.DataManager
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerSwapHandItemsEvent

class PlayerSwapHand : Listener {
	@EventHandler
	fun onSwapHand(e: PlayerSwapHandItemsEvent) {
		e.isCancelled = true
		val pd = DataManager.get(e.getPlayer()) ?: return
		pd.isPaused = !pd.isPaused
	}
}
